src/v/security/acl.h

```diff
@@ -291,6 +305,44 @@ class acl_principal final
       : _type(type)
       , _name(std::move(name)) {}
 
+    static acl_principal from_string(std::string_view principal) {
```

Comment:
since it's moving, it seems like an excellent opportunity to get it out of the header and into a compiled .cc file.